pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -696,24 +710,23 @@ func (adc *attachDetachController) processVolumesInUse(
 //
 //	if yes, the reconciler will attempt attach on the volume;
 //	if not (could be a dangling attachment), the reconciler will detach this volume.
-func (adc *attachDetachController) processVolumeAttachments() error {
+func (adc *attachDetachController) processVolumeAttachments(logger klog.Logger) error {
 	vas, err := adc.volumeAttachmentLister.List(labels.Everything())
 	if err != nil {
-		klog.Errorf("failed to list VolumeAttachment objects: %v", err)
+		logger.Error(err, "Failed to list VolumeAttachment objects")
 		return err
 	}
 	for _, va := range vas {
 		nodeName := types.NodeName(va.Spec.NodeName)
 		pvName := va.Spec.Source.PersistentVolumeName
 		if pvName == nil {
 			// Currently VA objects are created for CSI volumes only. nil pvName is unexpected, generate a warning
-			klog.Warningf("Skipping the va as its pvName is nil, va.Name: %q, nodeName: %q",
-				va.Name, nodeName)
+			logger.Info("Skipping the va as its pvName is nil", "node", klog.KRef("", string(nodeName)), "vaName", va.Name)
 			continue
 		}
 		pv, err := adc.pvLister.Get(*pvName)
 		if err != nil {
-			klog.Errorf("Unable to lookup pv object for: %q, err: %v", *pvName, err)
+			logger.Error(err, "Unable to lookup pv object", "pvName", *pvName)
```

Comment:
Sorry, missed this earlier: log this as `"pv", klog.KRef("", pvName)`.

Reply:
Correction: for consistency with "PVC", use "PV".

Reply:
done